### PR TITLE
perf: Improved element matching styles

### PR DESCRIPTION
**Pull request by devjiangzhou**
- [x]  Inherit style https://github.com/openwebf/webf/issues/108
- [x]  ID / Name Map Cache
- [x]  Improved element matching styles
- [x]  fix Variable & shorthands

### CODE REVIEW

**Review by andycall — APPROVED** [review]

**Comment by suyulin** [comment]
padding not support
```
<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="UTF-8" />
    <meta http-equiv="X-UA-Compatible" content="IE=edge" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Document</title>
  </head>
  <style>
    body{
        margin: 0;
        padding: 0;
    }
    :root {
      --padding-var: 50px;
    }
    .my-div {
      padding: 0 var(--padding-var);
      background-color: black;
      width: 50px;
      height: 50px;
    }
  </style>
  <body>
    <div class="my-div"></div>
  </body>
</html>

```

**Review by XGHeaven — CHANGES_REQUESTED** [review]

File: webf/lib/src/dom/document.dart [hunk 1]
```diff
@@ -166,7 +171,28 @@ class Document extends Node {
 
   dynamic getElementById(List<dynamic> args) {
     if (args[0].runtimeType == String && (args[0] as String).isEmpty) return null;
-    return QuerySelector.querySelector(this, '#' + args.first);
+    final elements = elementsByID[args.first];
+    if (elements == null || elements.isEmpty) {
+      return null;
+    }
+    if (elements.length == 1) {
+      return elements.last;
```
[the remark this answers is not in the document]
Comment (devjiangzhou) [reply webf/lib/src/dom/document.dart:179]: disconnectedCallback 已经处理了

File: webf/lib/src/dom/document.dart [hunk 2]
```diff
@@ -182,7 +208,7 @@ class Document extends Node {
 
   dynamic getElementsByName(List<dynamic> args) {
     if (args[0].runtimeType == String && (args[0] as String).isEmpty) return [];
-    return QuerySelector.querySelectorAll(this, '[name="${args.first}"]');
+    return elementsByName[args.first];
```
[the remark this answers is not in the document]
Comment (devjiangzhou) [reply webf/lib/src/dom/document.dart:211]: 同上

File: webf/lib/src/css/style_sheet.dart [hunk 3]
```diff
@@ -23,8 +23,6 @@ class CSSStyleSheet implements StyleSheet, Comparable {
 
   CSSStyleSheet(this.cssRules, {this.disabled = false, this.href});
 
-  // CSSStyleSheet(this.content, {this.disabled = false, this.href}) : cssRules = CSSParser(content).parseRules();
```
[the remark this answers is not in the document]
Comment (devjiangzhou) [reply webf/lib/src/css/style_sheet.dart:26]: 已经删除了

File: webf/lib/src/css/css_property_name.dart [hunk 4]
```diff
@@ -0,0 +1,1624 @@
...
+  VerticalAlign,
+  Visibility,
+  WhiteSpace,
+  Widows,
+  Width,
+  WordBreak,
+  WordSpacing,
+  WordWrap,
+  ZIndex,
+  WebkitAnimation,
```
[the remark this answers is not in the document]
Comment (devjiangzhou) [reply webf/lib/src/css/css_property_name.dart:173]: 删除了

File: webf/lib/src/css/css_property_name.dart [hunk 5]
```diff
@@ -0,0 +1,1624 @@
+/*
+ * Copyright (C) 2022-present The WebF authors. All rights reserved.
+ */
+
+enum CSSPropertyID {
```
[the remark this answers is not in the document]
Comment (devjiangzhou) [reply webf/lib/src/css/css_property_name.dart:5]: 等 CSS 这块儿参与的人多了再处理吧

**Comment by suyulin** [comment]
css  transform: translate(-50%, -50%); not support
```
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Document</title>
    <style>
        .div {
            width: 500px;
            height: 500px;
            position: relative;
            background-color: #7F583F;
          }
          .span {
            background: orange;
            position: absolute;
            top: 50%;
            left: 50%;
            transform: translate(-50%, -50%);
          }
    </style>
</head>
<body>
   <div class="div">
    <span class="span">居中</span>
   </div> 
</body>
</html>
```

**Comment by devjiangzhou** [comment]
> css transform: translate(-50%, -50%); not support
> 
> ```
> <!DOCTYPE html>
> <html lang="en">
> <head>
>     <meta charset="UTF-8">
>     <meta http-equiv="X-UA-Compatible" content="IE=edge">
>     <meta name="viewport" content="width=device-width, initial-scale=1.0">
>     <title>Document</title>
>     <style>
>         .div {
>             width: 500px;
>             height: 500px;
>             position: relative;
>             background-color: #7F583F;
>           }
>           .span {
>             background: orange;
>             position: absolute;
>             top: 50%;
>             left: 50%;
>             transform: translate(-50%, -50%);
>           }
>     </style>
> </head>
> <body>
>    <div class="div">
>     <span class="span">居中</span>
>    </div> 
> </body>
> </html>
> ```

单独提 PR 修吧

**Comment by yifei8** [comment]
https://github.com/openwebf/webf/issues/142

**Pull request merged**